test/web-service/__snapshots__/nginx-util.test.ts.snap

```diff
@@ -164,6 +168,142 @@ exports[`nginx-util > createConfigMap > Empty 1`] = `
 ]
 `;
 
+exports[`nginx-util > createConfigMap > Partitioned cookies config 1`] = `
+[
+  {
+    "apiVersion": "v1",
+    "data": {
+      "default.conf": "map $http_upgrade $connection_upgrade {
+    default "upgrade";
+    "" "";
+}
+
+upstream application {
+    server localhost:8080;
+    keepalive 256;
+}
+
+server {
+    listen 80;
+    server_name localhost;
+
+    gzip on;
+    gzip_comp_level 4;
+    gzip_types text/html text/plain text/css application/json application/x-javascript text/xml application/xml application/xml+rss text/javascript;
+
+    location / {
+        proxy_pass http://application;
+        proxy_http_version 1.1;
+
+        proxy_set_header Connection $connection_upgrade;
+        proxy_set_header Upgrade $http_upgrade;
+        proxy_set_header Host $host;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    }
+
+    location /livez {
+        access_log off;
+        add_header Content-Type text/plain;
+        return 200 'OK';
+    }
+
+    location /nginx_status {
+        stub_status on;
+        access_log off;
+        allow 127.0.0.1;
+        allow 10.0.0.0/8;
+        allow 172.16.0.0/12;
+        deny all;
+    }
+
+    location /api/oidclogin {
+      proxy_cookie_path / "/; Partitioned";
+    }
+}
+",
+    },
+    "kind": "ConfigMap",
+    "metadata": {
+      "labels": {
+        "prunable": "true",
+      },
+      "name": "test-nginx-config-c88fe926-b655fb47gd",
+    },
+  },
+]
+`;
+
+exports[`nginx-util > createConfigMap > Partitioned cookies config with multiple locations 1`] = `
+[
+  {
+    "apiVersion": "v1",
+    "data": {
+      "default.conf": "map $http_upgrade $connection_upgrade {
+    default "upgrade";
+    "" "";
+}
+
+upstream application {
+    server localhost:8080;
+    keepalive 256;
+}
+
+server {
+    listen 80;
+    server_name localhost;
+
+    gzip on;
+    gzip_comp_level 4;
+    gzip_types text/html text/plain text/css application/json application/x-javascript text/xml application/xml application/xml+rss text/javascript;
+
+    location / {
+        proxy_pass http://application;
+        proxy_http_version 1.1;
+
+        proxy_set_header Connection $connection_upgrade;
+        proxy_set_header Upgrade $http_upgrade;
+        proxy_set_header Host $host;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    }
+
+    location /livez {
+        access_log off;
+        add_header Content-Type text/plain;
+        return 200 'OK';
+    }
+
+    location /nginx_status {
+        stub_status on;
+        access_log off;
+        allow 127.0.0.1;
+        allow 10.0.0.0/8;
+        allow 172.16.0.0/12;
+        deny all;
+    }
+
+    location /api/oidclogin {
+      proxy_cookie_path / "/; Partitioned";
```

Comment:
question: Do these need the proxy settings suggested by the `/` location? I am assuming I need to add these other settings (such as `proxy_pass`).

Reply:
I did some quick testing - no, it won't work without the proxy stuff.
I thought we could nest the `location`s under `location /`, but proxy settings aren't inherited in Nginx: https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_set_header

So yes, the proxy stanzas will need to be duplicated under each `location`. I think it'll be best to extract them into a variable/file and inject were needed.